### PR TITLE
Update pairwise mapping for `SDPAFwdOp` outputs

### PR DESCRIPTION
**Pull request by Priya2698**
This PR:
- modifies the variables `cum_seq_q/k` to have root_domain as `{iN}` and logical domain as `{iN+1}`.
- updates the tests to validate all inputs from the forward pass. This excludes the scalar outputs `query_seq_len/key_seq_len` which will added in the follow-up PR.

### CODE REVIEW

**Comment by Priya2698** [comment]
!build

File: csrc/ops/composite.cpp [hunk 1]
```diff
@@ -471,16 +471,20 @@ SdpfaFwdResult sdpfa_fwd(
 
   // Create a new Tensorview for cum_seq_q, cum_seq_k of shape (N + 1)
   auto newForCumulativeSeq = [&]() -> TensorView* {
-    IterDomain* batch_id = ops::newOutputIterDomain({query_domain.front()});
-    batch_id = IterDomain::resize(
+    IterDomain* batch_id = ops::newOutputIterDomain({query_domain.front(), key_domain.front(), value_domain.front()});
```
Comment (naoyam) [inline csrc/ops/composite.cpp:474]: Why is this change?

`key_domain.front(), value_domain.front()}`

Comment (Priya2698) [reply csrc/ops/composite.cpp:474]: `ops::newOutputIterDomain` can use a series of iterdomains to construct a new iterdomain. If any of these are non-symbolic, the extents are set using that iterdomain. This is useful in resolving broadcast iterdomains.
In practice, this maybe unlikely so using just `query_domain.front()` is sufficient too.

This is similar to the construction of attention output and log_sumexp in L446-447, L461-462 which use the iterdomains corresponding to `N` from all inputs (q/k/v).

This was missing for `cum_seq_q/k`, so updating now.

**Review by naoyam — APPROVED** [review]
LGTM

File: tests/cpp/test_sdpa_node.cpp [hunk 2]
```diff
@@ -188,44 +223,54 @@ TEST_F(SDPATest, PairwiseRootDomainMap) {
...
-  fusion->addOutput(tvattn.output);
-
-  // Verify mapping between Q,K,V and attention output
+  addSdpaFwdOutputs(fusion.get(), output);
+
+  // Verify mapping between Q,K,V and all output
+  // Producers:
+  //   query = [N, H, L, E]
+  //   key = [N, H, S, E]
+  //   value = [N, H, S, Ev]
```
Comment (jjsjann123) [inline tests/cpp/test_sdpa_node.cpp:239]: nitpick: Looks like we support broadcasting on the batch dimension from the code changes above? do we actually have a test case for that where one of the entries have a broadcast batch dimension?

**Review by jjsjann123 — APPROVED** [review]

File: tests/cpp/test_sdpa_node.cpp (hunk 2 above)
Comment (Priya2698) [reply tests/cpp/test_sdpa_node.cpp:239]: Yes, broadcast is allowed on N, H.
I have a separate branch adding some checks enforcing `E = Ev` and the corresponding mapping, I will include it there.

Thanks for bringing this up.

Comment (jjsjann123) [reply tests/cpp/test_sdpa_node.cpp:239]: Feel free to add that in a separate PR if that works better~

**Comment by Priya2698** [comment]
!build

**Pull request merged**